OsmAnd/src/net/osmand/plus/myplaces/tracks/TracksSearchFilter.java

```diff
@@ -55,39 +54,57 @@ public TracksSearchFilter(@NonNull OsmandApplication app, @NonNull List<TrackIte
 		initFilters(app);
 	}
 
+	@SuppressWarnings("unchecked")
 	private void initFilters(@NonNull OsmandApplication app) {
 		recreateFilters();
-		DateCreationTrackFilter dateFilter = (DateCreationTrackFilter) getFilterByType(FilterType.DATE_CREATION);
-		if (dateFilter != null) {
-			long minDate = app.getGpxDbHelper().getTracksMinCreateDate();
-			long now = (new Date()).getTime();
-			dateFilter.setInitialValueFrom(minDate);
-			dateFilter.setInitialValueTo(now);
-			dateFilter.setValueFrom(minDate);
-			dateFilter.setValueTo(now);
-		}
-		LengthTrackFilter lengthFilter = (LengthTrackFilter) getFilterByType(FilterType.LENGTH);
-		if (lengthFilter != null) {
-			lengthFilter.setMaxValue((float) app.getGpxDbHelper().getTracksMaxDuration());
-		}
-		CityTrackFilter cityFilter = (CityTrackFilter) getFilterByType(FilterType.CITY);
-		if (cityFilter != null) {
-			cityFilter.setFullItemsCollection(app.getGpxDbHelper().getNearestCityList());
-		}
-		ColorTrackFilter colorsFilter = (ColorTrackFilter) getFilterByType(FilterType.COLOR);
-		if (colorsFilter != null) {
-			colorsFilter.setFullItemsCollection(app.getGpxDbHelper().getTrackColorsList());
-		}
-		WidthTrackFilter widthFilter = (WidthTrackFilter) getFilterByType(FilterType.WIDTH);
-		if (widthFilter != null) {
-			widthFilter.setFullItemsCollection(app.getGpxDbHelper().getTrackWidthList());
-		}
-		TrackFolderFilter folderFilter = (TrackFolderFilter) getFilterByType(FilterType.FOLDER);
-		if (folderFilter != null) {
-			folderFilter.setFullItemsCollection(app.getGpxDbHelper().getTrackFolders());
-			folderFilter.setCurrentFolder(currentFolder);
-		}
+		BackgroundThreadTask<Void> initFiltersTask = new BackgroundThreadTask<>(new BackgroundThreadTask.BackgroundThreadExecuteSource<Void>() {
```

Comment:
Probably BackgroundThreadTask needs some improvements because there are a few changes in code compared to simple AsyncTask